src/Stormpath.AspNet/Web/Http/Filters/StormpathCustomDataRequiredAttribute.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Stormpath.SDK;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http.Filters;
+
+namespace Stormpath.AspNet.Web.Http.Filters
+{
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true)]
+    public sealed class StormpathCustomDataRequiredAttribute : Attribute, IAuthenticationFilter
+    {
+        private readonly string key;
+        private readonly object value;
+
+        public StormpathCustomDataRequiredAttribute(string key, object value)
+        {
+            this.key = key;
+            this.value = value;
+        }
+
+        public async Task AuthenticateAsync(HttpAuthenticationContext context, CancellationToken cancellationToken)
+        {
+            if (!context.Principal.Identity.IsAuthenticated)
+            {
+                context.ErrorResult = new AuthorizationFailureResult("Not Authorized", context.Request);
+                return;
+            }
+
+            var account = context.Request.GetStormpathAccount();
+            if (account == null)
+            {
+                context.ErrorResult = new AuthorizationFailureResult("Not Authorized", context.Request);
+                return;
+            }
+
+            var customData = await account.GetCustomDataAsync();
+            if (customData == null)
+            {
+                context.ErrorResult = new AuthorizationFailureResult("Not Authorized", context.Request);
+                return;
+            }
+
+            if (customData[key] == null)

```

Comment:
This could be combined with the condition above: `if (customData?[key] == null)`